interface/typed_signature.py

```diff
@@ -20,6 +20,8 @@ def __init__(self, obj):
         self._type = type(obj)
         if isinstance(obj, (classmethod, staticmethod)):
             self._signature = signature(obj.__func__)
+        elif isinstance(obj, property):
```

Comment:
(We could actually get fancy and require compatibility for the triple of (`fget`, `fset`, and `fdel`), but I think starting with implicitly read-only properties is fine for now.)